tests/test_resolutions.py

```diff
@@ -570,12 +572,35 @@ def test_the_test(self) -> None:
         assert m.matches("Spam spam spam spam baked beans and spam")
         assert not m.matches("What do you mean Eugh?!")
 
+    def test_the_test_with_compile(self) -> None:
+        m = Matches(re.compile(r"([Ss]pam ?)+")).resolve()
+
+        assert m.matches("Spam spam spam spam baked beans and spam")
+        assert not m.matches("What do you mean Eugh?!")
```

Comment:
I personally wouldn't mind having this be part of the "test the test" test above, that's the pattern we've followed in all the other tests. But i don't mind it being a separate test.